src/Symfony/Bundle/TwigBundle/Tests/ContainerAwareRuntimeLoaderTest.php

```diff
@@ -11,28 +11,36 @@
 
 namespace Symfony\Bundle\TwigBundle\Tests;
 
+use Psr\Container\ContainerInterface;
 use Psr\Log\LoggerInterface;
-use Symfony\Component\DependencyInjection\ContainerInterface;
 use Symfony\Bundle\TwigBundle\ContainerAwareRuntimeLoader;
 
 class ContainerAwareRuntimeLoaderTest extends TestCase
 {
     public function testLoad()
     {
         $container = $this->getMockBuilder(ContainerInterface::class)->getMock();
-        $container->expects($this->once())->method('get')->with('foo');
+        $container->expects($this->once())->method('has')->with('FooClass')->willReturn(true);
+        $container->expects($this->once())->method('get')->with('FooClass');
 
-        $loader = new ContainerAwareRuntimeLoader($container, array(
-            'FooClass' => 'foo',
-        ));
+        $loader = new ContainerAwareRuntimeLoader($container);
         $loader->load('FooClass');
     }
 
     public function testLoadWithoutAMatch()
     {
         $logger = $this->getMockBuilder(LoggerInterface::class)->getMock();
         $logger->expects($this->once())->method('warning')->with('Class "BarClass" is not configured as a Twig runtime. Add the "twig.runtime" tag to the related service in the container.');
-        $loader = new ContainerAwareRuntimeLoader($this->getMockBuilder(ContainerInterface::class)->getMock(), array(), $logger);
+        $loader = new ContainerAwareRuntimeLoader($this->getMockBuilder(ContainerInterface::class)->getMock(), $logger);
         $this->assertNull($loader->load('BarClass'));
     }
+
+    /**
+     * @group legacy
+     * @expectedDeprecation The second argument of Symfony\Bundle\TwigBundle\ContainerAwareRuntimeLoader::__construct() is deprecated since version 3.3 and will be removed, passing something else than a Psr\Log\LoggerInterface instance will trigger an error in 4.0.
+     */
+    public function testSecondConstructorArgumentIsDeprecated()
+    {
+        $loader = new ContainerAwareRuntimeLoader($this->getMockBuilder(ContainerInterface::class)->getMock(), array());
```

Comment:
you also need to test that the existing behavior (using a mapping) keeps working. Copy the existing ``testLoad`` method for that.